### ISSUE TITLE
Removal of the 'rx_' prefix

### ISSUE THREAD

**Issue by larcus94** [ReactiveX/RxSwift#82]
What I was wondering is if there was a possibility of getting rid of the `rx_` prefix. I'm not gonna lie, I think it looks out of place in the Swift world. Thus this is really only of 'cosmetic' nature and not particularly high priority.

Either way, the pros of the `rx_` prefix is definitely discoverability and clarity. So a new naming convention would need to be at least similarly discoverable and clear. 

I was for example thinking about `observable`, so `rx_text` would become `observableText`. In my opinion, this is clear enough and looks a lot nicer.

Anyways, I just opened this issue to figure out what everyone else thinks about this. 
Do you prefer the `rx_` prefix? If not, what naming convention would you propose?

**Comment by bontoJR** [comment]
Thinking about it, I would propose to use `rxText` or `rxValue` that is more Swift oriented. Sticking with rx makes composition more clear than `observableText`, IMHO.

**Comment by larcus94** [comment]
`rxText` is already better. What's wrong with `textObservable`? It's clear what it returns and not ambiguous either. It's just not as discoverable.

**Comment by kzaher** [comment]
Hi @larcus94 ,

just to give people some context, we've had "flamewar" on twitter regarding this. @alcarvalho would be proud of us :)

Let me try to answer questions one by one. The reason why I'm writing this here is such a length is because we didn't have a discussion about this before, and I would like to capture how I feel about this. I would also like to have that discussions once preferably, so we don't waste a lot of time on it in the future.

Since we are focusing on RxSwift 2.0 and we can do breaking changes, I feel this issue is really important one in case we can find a better naming conventions practice.

I would also like to hear other people suggestions and alternatives.

First, I wouldn't use the term "wrong" anywhere where it comes to naming things. It's a matter of tradeoff.

These are the reason why we currently have `rx_text`

* you obviously need some way of reducing the chance of naming conflicts
* it would be great if you could easily find all of the extension methods that some framework adds to a specific class (that maybe looks like silly reason, but it's really important one IMHO)
* it should be consistent with ecosystem naming practices
* `text` was already used
* I would like for that mechanism to be generic, so we don't have to rethink names for each particular case, but it's pretty obvious what the name should be
* that quality should also help users of the library to find those methods, and have low cognitive overload while using it

I would ideally want to write something like

```
textField.rx.text
    >- map { t in ... }
```

Unfortunately didn't see that pattern being used anywhere else.

This is first result when you type "uibutton extension swift" https://github.com/Haneke/HanekeSwift/blob/master/Haneke/UIButton%2BHaneke.swift

I know that `rx_text` really screams that this is extension, but that's not necessarily a bad thing. You want to have that information when reviewing code. Your visual cortex doesn't have have doubts that it is seeing an extension method.

I have seen people writing extensions without prefix also, but IMHO that's bad practice for **large** projects for multiple reasons.

* naming collisions
* when using a method or doing a CR of those changes, you have no way of knowing is this built in method, or does it some through some third party SDK, and that is really important IMHO
* hurts discoverability

I wouldn't want for this to look like some kind of conspiracy of @bontoJR and me, but I really do think `rxText` is better then `textObservable`.

Let me try to explain why I think we can do better then `textObservable`.

It is true that apple platform has a very verbose naming conventions, but it has it's limits. For example:

```
optional func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat
```

it's not 

```
optional func tableView(tableView: UITableView, floatHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat
```

But on the other hand you have

```
@property (readonly) float floatValue;
@property (readonly) double doubleValue;
```

The reason why I would assume that is the case is because if you can conclude from the method name what type is it returning, then you don't want to encode that type into the name because it is redundant.

So in short :), if method arguments have semantics, then you don't encode type in name.

Text can only be `String`, height in UIKit does only probably make sense as `CGFloat`, value doesn't have any semantics at all, so reviewer would need a little help with that.

`textObservable` is actually not that precise. What would be really precise is `textStringObservable`, but that's unnecessary duplication IMHO.

`Observable` part IMHO doesn't really help code reviewer to figure out what is the type of observable, and is kind of redundant because almost every method in Rx returns some kind of observable.

If the purpose of `Observable` is to hint that we are using Rx, I think there are a lot conciser and preciser alternatives that give the same amount of information to the code reviewer.

`textObservable` also doesn't really help with discoverability of other methods because it uses a suffix.

These are the alternatives that I feel are acceptable for RxSwift 2.0.

* rxText - it is shortest, less conspicuous then `rx_text` and `rx.text` when doing CR, the distinction between framework namespace and real name is not that clear
* rx.text - it screams less then `rx_text` and more then `rxText`. You still have clear separation of namespace and real name. Haven't seen this used anywhere, so it could be weird for people to figure it out.
* rx_text - it screams most when doing CR, but it has been standard practice for a really long time
* some other alternative that we haven't thought of ...

@larcus94 let me know what do you think about those suggestions, and how would you order them by likeness, or do you have some suggestion other then `textObservable`?

I would also like are there any other ideas, and how do other people like those suggestions.

@bontoJR ?

Maybe also include @alcarvalho, @carlosypunto , @orta ....

**Comment by kzaher** [comment]
Oh, also @ashfurrow , @m00sey , @d6u, @shuoli84 .... I've missed probably a lot of people, but feel free to comment

**Comment by bontoJR** [comment]
Ouch, I started a long reply, then I decided to stop because it was too long to read, anyway, I think @kzaher is pretty clear about the decision why `rx_` is the current prefix.

I could go even deeper, but this would imply a lot of knowledge about what's behind data flow oriented programming, functional reactive programming (yes, the one of Conal), etc... All of the implementations and abstractions that I quoted have a certain name convention and all of them agree to ignore the implementation name of the abstraction they are relying on. The reason is to have a concise and clear name about the kind of data an observer is observing (sorry for the redundancy). Rx is about observing a value and a name of a variable should reflect that, that's why I would eventually stick for `rx_text` or support the `rxText` switch, that looks more swifty. :)

**Comment by orta** [comment]
I'd prefer to read a lot of `rxText`s over `rx_text`s for sure. 

The pattern of `button.rx.text` looks really appealing to me, if feels like if you typed `button.rx.`  the auto complete would show you all `rx`able methods in the drop down and could make life easier as a consumer in general.

**Comment by larcus94** [comment]
Yeah I was aware of the fact that `textObservable` is worse, I was just trying to come up with alternatives.
I like `rx.text` a lot. `rxText` is nice too, as long as it doesn't have an underscore really :)

**Comment by carlosypunto** [comment]
I sincerely feel comfortable With `rx_` prefix, IMHO rxText is good too

**Comment by shuoli84** [comment]
Since adopting rxSwift into a project already means mind shift, use >- and rx_ is purely ok for me. 
I have some concern on rx.text, may conflict on a property named "rx".
rx_text ~= rxText >> rx.text

**Comment by d6u** [comment]
I agree with @kzaher about we don't have to indicate the type in method names. Swift is a strong type language, type can be easily inferred.
I totally love the `xxx_` naming convention, because it's easier for eyes.

I wonder what's the implication of `rx.text`. Because `rx` as a property sounds like its reference can be saved and re-used later. Is this the desired behavior? Would that cause memory leak? I like to get answers for those questions instead of passing them to developers who use RxSwift.

**Comment by ashfurrow** [comment]
rxText seems nice to me

**Comment by m00sey** [comment]
This is a snippet of the ideology from RxBinding

> Observable factory method names is the plural of the verb (e.g., click --> clicks()). The verb should be  > in the present tense, regardless of the platform's use (e.g., selected -> selection). When there are  >multiple versions of the same verb, prefix with a qualifying noun or adjective that differentiates (e.g., click vs. long click, item selection vs. nothing selection).

Which in turn would lead to something along the lines of:

`username.rxTextChanges`
`login.rxClicks`
`datePicker.rxSelection`

More food for thought, the rest of the rationale for RxBinding is worth a read.

**Comment by kzaher** [comment]
@m00sey I'm looking at 

https://github.com/JakeWharton/RxBinding/blob/master/rxbinding/src/main/java/com/jakewharton/rxbinding/widget/RxTextView.java

It's weird to look at the same concept in different language :)

I like the naming for `login.rxTaps`? I'm assuming they would be taps since UIKit.

`username.rxTextChanges` kind of confuses me:

* it doesn't contain changes only, it also contains initial value
* it has different naming convention then `login.rxTaps`

If I were to read that method, from the description I would be sure that it didn't emit initial value, even though implementation says it does.

I'm trying to figure out why they don't use `username.rxTexts`? 

I think you've said something about being from Pittsburg, so if you know Jake Wharton I am really curios why they've made those decisions :)

@m00sey why is it `datePicker.rxSelection` instead of `datePicker.rxValues`?

I like the idea of plural form because it is a sequence of values, so it really is taps and texts.

**Comment by kzaher** [comment]
Over the weekend I've gave a closer look to `rxrs`, love what they've done there.

https://github.com/Reactive-Extensions/RxJS-DOM

```
var throttledInput = Rx.DOM.keyup(textInput);
```

One additional suggestion

```
rx.text(textField)
```

I just love what this communicates, there is no naming collisions, but all functions are inside the same namespace, so discovery is little easier. It also has 3 characters more then `rxText`.

I've talked with @bontoJR , and using plural form does have it's issues.

and what RxBindings does

bontojr [9:04 AM]
I do believe the current solution is the best :simple_smile:

bontojr [9:04 AM]
let me explain what I mean

bontojr [9:05 AM]
I checked some of the examples around the web in JS, Java, .NET, etc...

bontojr [9:06 AM]
and... I do believe that plural for sequences is misleading.

bontojr [9:07 AM]
example: a UITextField usually deliver a single text string

bontojr [9:07 AM]
on the other hand, if I use RxSwift for CoreData using the text field to return a search result

bontojr [9:08 AM]
I return a list of items from that fetch request

bontojr [9:08 AM]
so I would use like rxResults, because is a list

bontojr [9:08 AM]
even if a sequence of lists

bontojr [9:08 AM]
if I use plural for a textField, how I would represent a collection? rxTexts and then rxArrayResults?

Right now there really isn't any clear picture in my head what we'll end up with :(

But we have a month to figure it out.

**Comment by ninjazoete** [comment]
I read your discussion here and what cought my eye was this item.rx container for all rx-related capabilities - I think it is a really nice grouping approach because we would never have to remember / guess and rx api just search thorugh this rx structure.

After so many years of Obj-c it is hard now to move away (our minds ) from prefixes as they were required but encapsulating it inside rx will somehow let us name it without them and it will feel natural.

**Comment by d6u** [comment]
One thing to note for `Rx.DOM.keyup(textInput);` style is, the `textInput` is an original HTML element. If a library what to do `textInput.rx_keyup`, it have to modify the `HTMLElement` prototype to add the `rx_keyup` property/method for all elements, which will cause performance problem and people hate to modify prototype of builtin objects in JavaScript.

An alternative for RxJS would be `select(textInputDomSelector).rx_kekup`. However, RxJS is not focused as a DOM selection library. At the same time, there are tons of DOM selection alternatives like jQuery on the market, none of them is guaranteed to be loaded with RxJS. So `thirdPartyDomSelect(textInputDomSelector).rx_keyup` also won't work.

RxJS really have no choice but to use `Rx.DOM.keyup(textInput)` style to interact with original HTML element.

Does extension in Swift has the same performance penalty like modifying JavaScript prototype?

**Comment by ashfurrow** [comment]
I think that whatever approach we take needs to be evaluated primarily on two criteria:

- Is this idiomatic to Swift?
- Is this comfortable for iOS/OS X developers? 

I think rxText strikes a good balance between "feeling swift" and being familiar to developers. We are used to accessing properties on instances and getting things.

**Comment by m00sey** [comment]
+1 for idiomatic.

I guess my knee jerk objection to the singular rxText, is what, in part, Ash eludes to:

> We are used to accessing properties on instances and getting things

Text is the property, but the singular nature I find not to be true to what you're *actually* getting. You are not getting a singular text value, you are getting a sequence, which may, or may not contain changes in the value of that property. The only thing singular is the actual observable, and I think we've established we don't really want to be part of naming.

I would imagine that's how they arrived at textChanges - it may not be completely inline with the implementation, but it is a true representation of how we would use it.

`username.rxTextChanges >- subscribe`
```
u
us
use
user
user@
 ```

rxText to me is too close to the original property, and doesn't convey you're not getting a string back, but rather a sequence/observable. It's only after I map/filter/subscribe, would I start thinking in terms of the value of `text`.

> I'm trying to figure out why they don't use username.rxTexts

@kzaher I would imagine because it just doesn't read as very 'good' English.

**Comment by orta** [comment]
> we take needs to be evaluated primarily on two criteria:

> * Is this idiomatic to Swift?
> * Is this comfortable for iOS/OS X developers?

Shared understanding with the other libs should should be up there too, it's all nice and good to have it perfect in our world but not taking into account the greater ecosystem means veering off the road from the projects initial aims.

**Comment by bontoJR** [comment]
@ashfurrow is saying something right: 

> Is this idiomatic to Swift?
Is this comfortable for iOS/OS X developers?

but maybe we can try something new with eventually an alias (I know alias are usually bad for code maintenance, but it can be a initial necessary trade-off), considering that Swift is still very young? Avoiding to push it too far.

@m00sey: 
> rxText to me is too close to the original property, and doesn't convey you're not getting a string back, but rather a sequence/observable.

The `rx `before `Text` gives a good hint behind the type we are getting back, I think is clear we are getting an Observable for a text, so in this case we are observing a string value.

> It's only after I map/filter/subscribe, would I start thinking in terms of the value of text

That's also true, but Reactive Extensions are about composition. If you consider only transforming and filtering operators (in an imperative style), the name can be misleading, but what about composing operators? Merge, concat, merge, sample, etc... requires a clear name to understand what we are combining. An example would be to combine a text field with a slider, so we have the value of the both components combined. I would say that `rxText` combined with `rxValue`, using `combineLatest` would make very clear what is happening: it is a single text combined with a single value. That's an opinion of course, I read so much about Functional Programming and all the variants over the years, my brain can be out of focus. :)

From a pure mathematical standpoint, the most suitable solution would be the one @kzaher proposed as last one: `rx.text(textField)`, in this case is clear we are switching from an imperative style to a declarative one, that is in the nature of Reactive Extensions. On the other side discoverability would be hard. I would personally put the following solutions in order of preferences:

1. `textField.rx.text`
2. `textField.rxText`
3. `rx.text(textField)`

**Comment by dtartaglia** [comment]
Regardless of the name chosen, it's just a name, but the idea of using something like `textField.rx.text` is more than just a name variant, here you are mis-using a class as a namespace. It's a structural difference and not just a question of naming. I very much dislike using classes as namespaces. I think it muddles the concept. That's my negative critique, now for my positive contribution.

What are you getting back when you call the function that returns an Observable<String>? We tend to use plural when we are getting back more than one thing. So `strings` for an array of strings, `tableViewCells` for an array of UITableViewCells. In this case, the method only returns a single thing though... an observable string. It does *not* return a sequence, rather it transmits a sequence.

One thing that Objective-C taught me is not to be afraid of long method names. Go with the name that does the job and don't worry so much about character count. In this case, the method is returning an observable on the textField's text attribute. The obvious name then is `textObserverable`. Another idea is that the object returned transmits the current value of the `text` field, so `currentText` sounds good to me as well.

Just my $0.02. :-)

**Comment by bontoJR** [comment]
@dtartaglia :
> Regardless of the name chosen, it's just a name, but the idea of using something like textField.rx.text is more than just a name variant, here you are mis-using a class as a namespace. It's a structural difference and not just a question of naming. I very much dislike using classes as namespaces. I think it muddles the concept. That's my negative critique, now for my positive contribution.

You are definitely right about using a class as namespace could generally lead to misunderstandings, but considering the kind of switch we are doing from the imperative style to a declarative one every time we use Rx, I don't think this would be a major issue. The namespace in this case indicates how big is the switch. With the namespace I would also say we are passing from the current imperative space, to the declarative.

In a first instance I was thinking that this is a just a matter of naming a thing, but in the Rx space is way more than this... having the wrong name convention would definitely mislead newcomers and confuse their mind. If we can have a very strong, clear rule, we can then document it and explain how we came up with that solution so all related libraries are going to follow the same pattern and be easy to use for a developer habit to RxSwift.

**Comment by kzaher** [comment]
Hi guys,

I don't think anybody will be satisfied with what ever solution we come up with, including myself (I think that I'll need to be satisfied with my 4th ranking from case of an ideal sandbox world where I have infinitely fast fingers).

I don't think we should choose a solution to be anyones first choice but rather a solution that is economical, is not horrible to anybody, and could pass as kind of idiomatic to platform.

If I were to choose a solution, but never have to write a line of rx code, I would probably choose `rx.text(textField)`. It's really a perfect solution IMHO, but I think people would be shocked and send me to hell, and not in that order necessarily :)

My second options would be `textField.rx.text`. I think people would find plenty of reasons to dislike it. Either because of "namespace" hacks (even though we need namespaces),  because of need for additional proxy object (people will write their own methods so having to write additional object is tedious), or because of non idiomatic code (even though it's not yet clear what is swift idiomatic extension code).

My third option was old `rx_text`. That didn't go well :) Probably only me and @d6u like it :)

My fourth option is `rxText` and I really feel this has biggest chance of being a winner. I don't think I've seen anyone said to be horrible.

I've tried to explain in great length why I don't feel `textObservable` is a good choice for this :), but I guess all mentioned reasons could have various weights to different people. And I think that's fair, but IMHO we have a possibly better alternative.

@d6u I don't think there is some significant performance penalty what ever solution we choose, they should all be similar (except for compile time complexity). Nice feedback regarding js pro/cons :+1:.

@m00sey I don't think you are wrong when you say that plural would suit it well (I've said I agreed), but I see it as being highly impractical (I think that's probably the reason why they've added always plural `Changes`). 

I have no idea how I would write core data entity observer. `entitiess` or `entitiesChanges`? I don't think that it is practical for naming algorithm to include changing plurality of the noun. That is really risky IMHO, not to mention internationalization issues. Not all rx code people write is necessarily in english.

And yes, I could find philosophical reasons why I could justify adding "changes". Like initial value is just a replay of last change from past, but I'm sure lot of people will find this reasoning unnatural, and not to mention it confuses me even though I know the subject pretty well.

I find it also extremely important that method name isn't misleading and I have to go into source code of the method itself on a new platform to be sure what it's doing.

I have obviously changed my opinions after hearing this discussion, and maybe I'll change it again if I hear some additional arguments, but anyway, these are my conclusions so far.

`rxText` has probably got most push back from @m00sey , @dtartaglia . I'm curios do you think it would be an absolutely horrible solution, or an acceptable tradeoff?

I think @larcus94 was fine with it?

**Comment by dtartaglia** [comment]
Frankly, I had no problem with `rx_text` nor do I have a problem with `rxText`. Both are fine solutions. Like I said, the only think I thought was a really bad idea, and the only reason I jumped into the conversation, was because I intensely disliked the idea of using a class as a namespace.

`rxText` is great.

**Comment by orta** [comment]
From the sidelines, I've been a fan of `thing.rx.text` - it marks the distinction really well between code domains, and is really easy to know all of the rx functions that the class supports. In a way,  the prefixes in `rx_XX` / `rxXX` are emulating a faux-namespace, which this accurately provides.

**Comment by larcus94** [comment]
`rx.text` looks very clean and swift-esque but must be so awkward to implement IMO. I'm perfectly fine with `rxText` :+1:

**Comment by m00sey** [comment]
Since adhering to plurality seems out, I think @orta nails the nail on the head as far as what we're trying to achieve with `rx_`, although there concerns with discovery?

As a final side note, the reason I'm so hooked on the plurality is that was *how* I was introduced to Rx as: 

> a library for composing asynchronous and event-based programs by using observable sequences

And the depiction of where it falls in the realm of tools, we as engineers have to us. 

![screen shot 2015-08-10 at 14 56 30](https://cloud.githubusercontent.com/assets/180035/9180365/06851a0a-3f70-11e5-9912-203ab4afc8a0.png)

**Comment by ninjazoete** [comment]
@larcus94 that is why I am all for rx container from the begining of this discussion. Prefixing a property with "rx" is a kind of fake namespace which only has one purpose - discoverability. Consider a control that has an *echo* property and now by hipotetical conention of RxSwift you can assume that you just have to write control.rx.echo just to get to the observable.

**Comment by d6u** [comment]
For me, `rx.text(textField)` > `rx_text` > `rxText` >>>>> `.tx.text` since editor should be the one who does a better job to auto complete a piece of code.

**Comment by kzaher** [comment]
@m00sey I think when speaking imperatively, it is `texts`, when speaking declaratively, it is `text`. It's just a matter of vista.

This is how `rx` could be implemented :) Tried it just now and it works :)

```
typealias NSTextFieldExtensions = NSTextField

extension NSTextField {
    var rx: NSTextFieldExtensions {
        return self
    }
}

extension NSTextFieldExtensions {
    var text: Observable<String> {
        return rx_text
    }
}
```

@d6u I think that what we actually have is a extension namespace issue here. Autocomplete working/not working is just a consequence do you fight the system.

**Comment by kzaher** [comment]
This is example of code with and without `.`. So we can compare then in context. I think they both look ok.

```
let sum = combineLatest(a.rx.text, b.rx.text) { (a: String, b: String) -> (Int, Int) in
            return (Int(a) ?? 0, Int(b) ?? 0)
        }
```

```
let sum = combineLatest(a.rxText, b.rxText) { (a: String, b: String) -> (Int, Int) in
            return (Int(a) ?? 0, Int(b) ?? 0)
        }
```

**Comment by ninjazoete** [comment]
@kzaher I tried your solution for rx grouping and if you put there e.g. UITextField it won't work as you can't overload the property that already has a certain getter and/or setter or in case of pure swift : **"Invalid redeclaration of <name>"**

Going further if you group it like that, returning self will make you go in autocomplete through all of the base type posibilities anyway :( so I am afraid this is not possible to map it 1 : 1 without creating a real rx container in each rx-supporting class.

**Comment by kzaher** [comment]
@ninjazoete Sry, my bad. But this should work, if I haven't messed up something again :)

```
protocol RxExtensionType {
    typealias Type
    
    var value: Type { get }
}

struct RxExtension<T> : RxExtensionType {
    typealias Type = T
    
    let value: Type
    init(_ value: Type) {
        self.value = value
    }
}

extension NSTextField {
    var rx: RxExtension<NSTextField> {
        return RxExtension(self)
    }
}

extension RxExtensionType where Type == NSTextField {
    var text: Observable<String> {
        return self.value.rx_text
    }
}
```

**Comment by bontoJR** [comment]
@orta 
> From the sidelines, I've been a fan of thing.rx.text - it marks the distinction really well between code domains, and is really easy to know all of the rx functions that the class supports.

Great answer! :+1:

**Comment by kzaher** [comment]
This is becoming interesting :)

https://github.com/SwiftBond/Bond/pull/125
https://github.com/ReactiveCocoa/ReactiveCocoa/pull/2189/files

are all going for `xxx_` :)

How can we find some objective measure what does Swift idiomatic mean?

**Comment by lfaoro** [comment]
I believe at some point you should go with your hearth and what you feel is mostly correct.
The general consensus seem to go towards `rxText` and it's not a bad thing.
`rx_` and `rx.` give more separation agreed and you're more inclined to type that and wait to see what autocomplete has to offer but it doesn't look clean nor Swifty.

I guess we could go with `rxElement` for version 2 and see what happens then.

As @ashfurrow mentioned:
- Is this idiomatic to Swift? | Absolutely
- Is this comfortable for iOS/OS X developers? | By the consensus shown in the comments, sure!

/cc @kzaher @bontoJR

**Comment by carlosypunto** [comment]
In a previous post I did not feel bad with camel-cased version. But after having used it in the early 2.0 alphas, body asks me `rx_` prefix far read the code. I find the `rx_` prefix right at all and I think it is a rather common practice in other libraries and recommended in certain circles as Ray Wenderlich tutorials.

 I do not likethe `think.rx.` prefix because it requires adding unnecessary code to the library

IMHO I retain the `rx_` prefix

**Comment by bontoJR** [comment]
So, I guess we would stick with `rx_`, what's the status @kzaher, can we close this? 
So we can focus on GTD on the feature side? :)

**Comment by kzaher** [comment]
Hi,

I think `rxXxxx` would be nice, but don't see any clear benefit over `rx_`, and there is a lot of people using `rx_` now :)

But I'm keeping my mind open :) 

I would like to see clearly in what direction is ecosystem moving. So far I see that other frameworks are actually moving to `xx_xx` format, but I'll keep my eyes open.

We plan to release RxSwift in about 2 months, so plenty of more time to make that naming change if we see some clear direction.

I don't mind leaving this open for now, don't think it hurts anyone, but yeah, wouldn't want to make this change if we won't see any clear benefit.

**Comment by lfaoro** [comment]
You mean Rx_Swift :-1:

**Comment by kzaher** [comment]
![](http://cdn.meme.am/instances/500x/63302021.jpg)

**Comment by lfaoro** [comment]
:+1: :smile:

**Comment by muukii** [comment]
rx. is great!

**Comment by gregpardo** [comment]
I like the current system. rxText seems good but I kind of like the fact that 'text' looks the same as 'text' in the rx_ version. And it does look like most the reactive libraries use a similar syntax which is a good reason to not switch it. If it's not broke, fixing it is a waste of resources that could be put into another area of the project.

**Comment by kzaher** [comment]
I think we should leave `rx_` for 2.0 version. 

If there are some new conclusions before 3.0 is released, we can consider this for 3.0.
